node/status_node.go

```diff
@@ -178,14 +180,40 @@ func (n *StatusNode) setupRPCClient() (err error) {
 }
 
 func (n *StatusNode) discoveryEnabled() bool {
-	return n.config != nil && !n.config.NoDiscovery && n.config.ClusterConfig != nil
+	return n.config != nil && !n.config.NoDiscovery && n.config.ClusterConfig != nil && len(n.config.EnabledDiscoveries) != 0
 }
 
 func (n *StatusNode) startDiscovery() error {
-	n.discovery = peers.NewDiscV5(
-		n.gethNode.Server().PrivateKey,
-		n.config.ListenAddr,
-		parseNodesV5(n.config.ClusterConfig.BootNodes))
+	if len(n.config.EnabledDiscoveries) != 1 {
+		return errors.New("only one discovery can be used (will be allowed to use more in next change)")
+	}
+	switch dtype := n.config.EnabledDiscoveries[0]; dtype {
+	case discovery.EthereumV5:
+		n.discovery = discovery.NewDiscV5(
+			n.gethNode.Server().PrivateKey,
+			n.config.ListenAddr,
+			parseNodesV5(n.config.ClusterConfig.BootNodes))
+	case discovery.RendezvousV1:
+		if len(n.config.ClusterConfig.RendezvousNodes) == 0 {
+			return errors.New("rendezvous node must be provided if rendezvous discovery is enabled")
+		}
+		maddrs := make([]ma.Multiaddr, len(n.config.ClusterConfig.RendezvousNodes))
+		for i, addr := range n.config.ClusterConfig.RendezvousNodes {
```

Comment:
I think we can move this `for` to a helper function.

Reply:
sure i can split